test/gtest/hello_test.cpp

```diff
@@ -0,0 +1,29 @@
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/solver_id.hpp>
+
+//Todo: Reduced duplicated code from cache.cpp
+//E.g. declare extern check_cache() in cache. cpp and declare dependency on cache.cpp?
+
+// Demonstrate some basic assertions.
+//Todo: Include a simple task
+//TEST_F()?: use the same data configuration for multiple tests
+
+TEST(HelloTest, BasicAssertions)
+{
+
+    // Expect two strings not to be equal.
+    EXPECT_STRNE("hello", "world");
+    // Expect equality.
+    EXPECT_EQ(7 * 6, 42);
+
+    // Check if we can access MIOpen internals
+    auto idx = 0;
+    for(const auto& solver_id :
+        miopen::solver::GetSolversByPrimitive(miopen::solver::Primitive::Convolution))
+    {
+        std::ignore = solver_id;
+        ++idx;
+    }
+    EXPECT_GT(idx, 0);
```

Comment:
Why cant we write `EXPECT(idx > 0)`?

Reply:
But will `EXPECT_TRUE` decompose the conditions(so you can see what value of `idx` was in this case)? Thats how it works in Catch, Boost.Test and even in migraphx's [test.hpp](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/generate.hpp#L72) header.

Reply:
> But will `EXPECT_TRUE` decompose the conditions(so you can see what value of `idx` was in this case)? Thats how it works in Catch, Boost.Test and even in migraphx's [test.hpp](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/generate.hpp#L72) header.

@pfultz2 Seems EXPECT_TRUE can output error message, e.g. below. Will update the shortly 
EXPECT_TRUE(marker == 1) << "test_activation failed";